src/historywork/GetAndUnzipRemoteFileWork.cpp

```diff
@@ -154,4 +162,15 @@ GetAndUnzipRemoteFileWork::validateFile()
 
     return true;
 }
+
+std::shared_ptr<HistoryArchive>
+GetAndUnzipRemoteFileWork::getArchive() const
+{
+    if (mGetRemoteFileWork &&
+        mGetRemoteFileWork->getState() == BasicWork::State::WORK_SUCCESS)
```

Comment:
Do we need the condition on work success? if we can't even talk to the archive, it can be a problem with the archive, right?

Reply:
If we can't talk to the archive, we'll get the log in `GetRemoteFileWork`. I kept the success condition here, so that we only log in the upstream works if the actual validation fails.